server/src/routes/assignment.ts

```diff
@@ -0,0 +1,126 @@
+import { controllerDispatcher } from '../middleware/errorHandler';
+
+export const router: Router = Router();
+
+/**
+ * @swagger
+ * definitions:
+ *   AddAssignment:
+ *     type: object
+ *     description: Assignment Information for adding a new assignment.
+ *     properties:
+ *       courseInstanceId:
+ *         type: number
+ *         description: Course instance ID to which the assignment belongs to.
+ *       name:
+ *         type: string
+ *         description: Assignment name.
+ *       executionDate:
+ *         type: string
+ *         description: Date when the assignment is completed (e.g. deadline date or exam date).
+ *       expiryDate:
+ *         type: string
+ *         description: Date when the assignment expires.
+ *   Assignment:
+ *     description: Current assignment Information.
+ *     allOf:
+ *       - type: object
+ *         properties:
+ *           id:
+ *             type: number
+ *             required: true
+ *             description: Assignment ID.
+ *       - $ref: '#/definitions/AddAssignment'
+ */
+
+/**
+ * @swagger
+ * /v1/assignment:
+ *   post:
+ *     tags: [Assignment]
+ *     description: Add a new assignment.
+ *     requestBody:
+ *       description: New assignment data.
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/definitions/AddAssignment'
+ *     responses:
+ *       200:
+ *         description: New assignment succesfully added.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Assignment'
+ *       400:
+ *         description: Creation failed, due to malformed/missing parameters.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       404:
+ *         description: Course instance does not exist.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ */
+router.post(
+  '/v1/assignment',
+  express.json(),
+  handleInvalidRequestJson,
+  controllerDispatcher(addAssignment)
+);
+
+/**
+ * @swagger
+ * /v1/assignment/{assignmentId}:
+ *   put:
+ *     tags: [Assignment]
+ *     description: Update existing assignment.
+ *     requestBody:
+ *       description:  Assignment data to be updated.
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/definitions/Assignment'
+ *     responses:
+ *       200:
+ *         description: Assignment updated succesfully.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Assignment'
+ *       400:
+ *         description: Creation failed, due to malformed/missing parameters.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       404:
+ *         description: Course instance does not exist.
```

Comment:
422 might be a better code in this case, at least when the instance ID is passed in the body. But this again is something that should be discussed later.